internal/metrics/proxy_metrics.go

```diff
@@ -0,0 +1,296 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"github.com/milvus-io/milvus/internal/util/typeutil"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	SuccessLabel   = "success"
+	FailLabel      = "fail"
+	TotalLabel     = "total"
+	CacheHitLabel  = "hit"
+	CacheMissLabel = "miss"
+)
+
+var (
+	// ProxyDmlChannelTimeTick counts the time tick value of dml channels
+	ProxyDmlChannelTimeTick = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "dml_channels_time_tick",
+			Help:      "Time tick of dml channels",
+		}, []string{"client_id", "pchan"})
+
+	// ProxySearchCount record the number of times search succeeded or failed.
+	ProxySearchCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "search_counter",
+			Help:      "The number of times search succeeded or failed",
+		}, []string{"client_id", "status"})
```

Comment:
Would it be better to keep a metric and then use a type to differentiate search/query?

Reply:
add a label named collection_id